tiny_dnn/core/kernels/recurrent_cell_op.h

```diff
@@ -0,0 +1,97 @@
+   information.
+
+    LICENSE
+
+    The BSD 3-Clause License
+
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright notice,
+   this
+      list of conditions and the following disclaimer.
+
+    * Redistributions in binary form must reproduce the above copyright notice,
+      this list of conditions and the following disclaimer in the documentation
+      and/or other materials provided with the distribution.
+
+    * Neither the name of tiny-dnn nor the names of its
+      contributors may be used to endorse or promote products derived from
+      this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+    AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+    IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+   ARE
+    DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+    FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+    DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+    SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+   LIABILITY,
+    OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+   USE
+    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+#pragma once
+
+#include "tiny_dnn/activations/activation_layer.h"
+#include "tiny_dnn/core/framework/op_kernel.h"
+#include "tiny_dnn/core/kernels/recurrent_cell_op_internal.h"
+
+namespace tiny_dnn {
+
+class RecurrentCellOp : public core::OpKernel {
+ public:
+  explicit RecurrentCellOp(const core::OpKernelConstruction &context)
+    : core::OpKernel(context) {}
+
+  void compute(const core::OpKernelContext &context) override {
+    auto params = OpKernel::params_->recurrent_cell();
+
+    // incomimg/outcoming data
+    const tensor_t &in_data = context.input(0);
+    const tensor_t &prev_h  = context.input(1);
+    const tensor_t &U       = context.input(2);
+    const tensor_t &W       = context.input(3);
+    const tensor_t &V       = context.input(4);
+    const tensor_t *bias    = params.has_bias_ ? &context.input(5) : nullptr;
+    const tensor_t *c       = params.has_bias_ ? &context.input(6) : nullptr;
+    tensor_t &out_data      = context.output(0);
+    tensor_t &next_h        = context.output(1);
+
+    // initialize outputs
+    fill_tensor(out_data, float_t{0});
+    fill_tensor(next_h, float_t{0});
+
+    // call the algorithm depending  on the selected engine type
+
+    const core::backend_t engine = context.engine();
+
+    if (engine == core::backend_t::internal || engine == core::backend_t::avx) {
```

Comment:
I don't think this `if` statement is needed since what we have now is `recurrent_cell_op_internal` function. We can't choose other options.